src/AlevinUtils.cpp

```diff
@@ -246,6 +259,25 @@ namespace alevin {
       return pt.umi_geo.extract_tag(read1, read2, umi);
     }
     template <>
+    bool extractUMI<apt::CustomGeo>(std::string& read1,
+                                 std::string& read2,
+                                 apt::CustomGeo& pt,
+                                 std::string& umi){
+      std::string um = "";
```

Comment:
So I think we could improve things a little bit here, potentially.  I think if we are declaring this string as such in each function call, we are allocating a string each time.  If the string remains small, we benefit from SSO (small string optimization), but it might still be faster (and safer) to have e.g. a member variable of `apt::CustomGeo` that we use as a cache for this, rather than each time allocating a new string.